ledger/complete/wal/checkpointer.go

```diff
@@ -773,50 +776,75 @@ func readCheckpointV5(f *os.File) ([]*trie.MTrie, error) {
-	cmdFileName := "/bin/dd"
-
-	// If /bin/dd isn't found, then try /usr/bin/dd (OpenSUSE Leap, etc.)
-	_, err := os.Stat(cmdFileName)
-	if os.IsNotExist(err) {
-		cmdFileName = "/usr/bin/dd"
-		_, err := os.Stat(cmdFileName)
-		if os.IsNotExist(err) {
-			return fmt.Errorf("required program dd not found in /bin/ and /usr/bin/")
+// EvictAllCheckpointsFromLinuxPageCache advises Linux to evict all checkpoint files
+// in dir from Linux page cache.  It returns list of files that Linux was
+// successfully advised to evict and first error encountered (if any).
+// Even after error advising eviction, it continues to advise eviction of remaining files.
+func EvictAllCheckpointsFromLinuxPageCache(dir string, logger *zerolog.Logger) ([]string, error) {
+	var err error
+	matches, err := filepath.Glob(filepath.Join(dir, checkpointFilenamePrefix+"*"))
+	if err != nil {
+		return nil, fmt.Errorf("failed to enumerate checkpoints: %w", err)
+	}
+	evictedFileNames := make([]string, 0, len(matches))
+	for _, fn := range matches {
+		base := filepath.Base(fn)
+		if !strings.HasPrefix(base, checkpointFilenamePrefix) {
+			continue
+		}
+		justNumber := base[len(checkpointFilenamePrefix):]
+		_, err := strconv.Atoi(justNumber)
+		if err != nil {
+			continue
+		}
+		evictErr := evictFileFromLinuxPageCacheByName(fn, false, logger)
+		if evictErr != nil {
+			if err == nil {
+				err = evictErr // Save first evict error encountered
+			}
+			if logger != nil {
+				logger.Warn().Msgf("failed to evict file %s from Linux page cache: %s", fn, err)
+			}
+			continue
 		}
+		evictedFileNames = append(evictedFileNames, fn)
 	}
+	// return the first error encountered
+	return evictedFileNames, err
+}
 
-	// Remove some special chars from fileName just in case.
-	// Regex would be shorter but not as easy to read.
-	s := strings.ReplaceAll(fileName, " ", "")
-	s = strings.ReplaceAll(s, ";", "")
-	s = strings.ReplaceAll(s, "$", "")
-	s = strings.ReplaceAll(s, "|", "")
-	s = strings.ReplaceAll(s, ">", "")
-	s = strings.ReplaceAll(s, "<", "")
-	s = strings.ReplaceAll(s, "*", "")
-
-	_, err = os.Stat(s)
-	if os.IsNotExist(err) {
-		return fmt.Errorf("sanitized filename %s does not exist", s)
+// evictFileFromLinuxPageCacheByName advises Linux to evict the file from Linux page cache.
+func evictFileFromLinuxPageCacheByName(fileName string, fsync bool, logger *zerolog.Logger) error {
+	f, err := os.Open(fileName)
+	if err != nil {
+		return err
 	}
+	defer f.Close()
 
-	cmd := exec.Command(cmdFileName, "if="+s, "iflag=nocache", "count=0")
+	return evictFileFromLinuxPageCache(f, fsync, logger)
+}
 
-	if logger != nil {
-		logger.Info().Msgf("run %q to drop file from OS file cache", cmd.String())
+// evictFileFromLinuxPageCache advises Linux to evict a file from Linux page cache.
+// A use case is when a new checkpoint is loaded or created, Linux may cache big
+// checkpoint files in memory until evictFileFromLinuxPageCache causes them to be
```

Comment:
```suggestion
// checkpoint files in memory until evictFileFromLinuxPageCache is called to force them to be
```

Reply:
`evictFileFromLinuxPageCache` calls `posix_fadvise` under the hood.  And `posix_fadvise` docs say that it's not binding advice and merely constitutes an expectation on behalf of the application.

In benchnet tests, the file is always removed from Linux page cache after calling `evictFileFromLinuxPageCache`.